### ISSUE TITLE
Validation script reporting near-perfect results for mismatched model and dataset

### ISSUE THREAD

**Issue by bj0rndan** [ultralytics/ultralytics#14476]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Val

### Bug

The validation script in YOLOv8 is reporting near-perfect results (mAP50 and mAP50-95 of 0.995) across all classes, even in scenarios where it should not. Using model.val() method it occurs when:

1. Validating a model on its original training dataset, despite initial metrics showing different, inferior performance.
2. **Using a model trained on separate classes to validate on a dataset with grouped classes that don't match the training classes.**
3. Changing validation parameters like confidence and IoU do not affect the "perfection" of the results

This behavior suggests a critical bug in the validation process, as it's reporting unrealistically high performance metrics even when the model and dataset are intentionally mismatched. This issue could lead to false confidence in model performance and needs urgent attention.

This are the model results after training logged with CometML:
![image](https://github.com/user-attachments/assets/044dc617-e80e-4490-ae2d-2fdde432487b)

This are the same model results using:
```
metrics = model.val(data="path_to_my_data",
                        save_json=True,
                        save_hybrid=True,
                        plots=True,
                        imgsz=640,
                        conf=0.25,
                        iou=0.6,
                        device=0)
```
![image](https://github.com/user-attachments/assets/49cf4286-0a18-469a-9387-aba5ad1469bc)

When changing validation parameters of confidence and IoU should make the model to alter the perfomance metrics. However the result is still the same:
```
conf=0.99,
iou=0.99
```
![image](https://github.com/user-attachments/assets/9c10594b-6a4f-46c4-be19-9921158ade40)

### Environment

Ultralytics YOLOv8.2.23 🚀 Python-3.10.14 torch-2.3.0+cpu CPU (13th Gen Intel Core(TM) i9-13900K)
Setup complete ✅ (32 CPUs, 31.7 GB RAM, 645.4/930.9 GB disk)

OS                  Windows-10-10.0.19044-SP0
Environment         Windows
Python              3.10.14
Install             pip
RAM                 31.75 GB
CPU                 13th Gen Intel Core(TM) i9-13900K
CUDA                None

matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.3.0>=1.8.0
torchvision         ✅ 0.18.0>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

### Minimal Reproducible Example

```
metrics = model.val(data="path_to_my_data",
                        save_json=True,
                        save_hybrid=True,
                        plots=True,
                        imgsz=640,
                        conf=0.25,
                        iou=0.6,
                        device=0)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
Because you are using `save_hybrid=True`.

@glenn-jocher Can an warning be added for this?

**Comment by bj0rndan** [comment]
Thank you for your response.
I think I missunderstood the purpose of this. Could you please explain?

save_hybrid | bool | False | If True, saves a hybrid version of labels that combines original annotations with additional model predictions.

> Because you are using `save_hybrid=True`.
> 
> @glenn-jocher Can an warning be added for this?

**Comment by Y-T-G** [comment]
It does what it says. But it breaks the mAP calculation.

**Comment by glenn-jocher** [comment]
Created https://github.com/ultralytics/ultralytics/pull/14484 to warn on this

**Comment by Y-T-G** [comment]
@glenn-jocher Oops, didn't see. I created one too just now.
